src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/CodeDirectiveFormattingTest.cs

```diff
@@ -689,6 +683,52 @@ protected override async Task OnInitializedAsync()
         forecasts = await ForecastService.GetForecastAsync(DateTime.Now);
     }
 }
+");
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/aspnetcore/issues/34320")]
+        public async Task CodeBlock_ObjectCollectionArrayInitializers()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@code {
+    public List<object> AList = new List<object>()
+    {
+        new
+        {
+            Name = ""One"",
+            Goo = new
+            {
+                First = 1,
+                Second = 2
+            },
+            Bar = new string[] {
+                ""Hello"",
+                ""There""
+            }
+        }
+    };
+}
+",
+expected: @"@code {
```

Comment:
Maybe explicitly call out that having no changes at all to the C# is the expected result?